iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -130,13 +131,14 @@ public void initialize(@Nullable Configuration configuration, Properties serDePr
       }
     }
 
-    this.projectedSchema = projectedSchema(configuration, serDeProperties.getProperty(Catalogs.NAME), tableSchema);
+    this.projectedSchema =
+        projectedSchema(configuration, serDeProperties.getProperty(Catalogs.NAME), tableSchema, jobConfs);
 
     // Currently ClusteredWriter is used which requires that records are ordered by partition keys.
     // Here we ensure that SortedDynPartitionOptimizer will kick in and do the sorting.
     // TODO: remove once we have both Fanout and ClusteredWriter available: HIVE-25948
-    HiveConf.setIntVar(configuration, HiveConf.ConfVars.HIVEOPTSORTDYNAMICPARTITIONTHRESHOLD, 1);
-    HiveConf.setVar(configuration, HiveConf.ConfVars.DYNAMICPARTITIONINGMODE, "nonstrict");
+    jobConfs.put(HiveConf.ConfVars.HIVEOPTSORTDYNAMICPARTITIONTHRESHOLD.varname, "1");
+    jobConfs.put(HiveConf.ConfVars.DYNAMICPARTITIONINGMODE.varname, "nonstrict");
```

Comment:
We now popularte `jobConfs` instead of `configuration`. Would it be a problem for `configuration` to miss out these settings when we return to the caller of this method?

Reply:
I don't think there is any problem there, the caller isn't explicitly passing, it is {{SessionState.getSessionConf()}}, and ideally the values which we are setting aren't of any use to session & all other SerDe use it as a Read-Only configuration.
And moreover I don't think we should change the session conf for no reason here, not sure if it can even bother further queries or so...